modules/ximgproc/src/find_ellipses.cpp

```diff
@@ -0,0 +1,2012 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "precomp.hpp"
+#include <opencv2/core.hpp>
+#include <unordered_map>
+
```

Comment:
to compile on windows i added 
`#include <numeric>`

Reply:
i tried your sample find_ellipses.cpp
here is the result. i wonder if your algorithm finds circles or not. or can find circles ?
![result](https://user-images.githubusercontent.com/12151541/183935301-759ffddd-83c7-42a9-94f1-3465e2e29459.jpg)

Reply:
Thank you very much for your reply, our method can find out circles in images.

However, for this schematic, it is fatal for all the arc-based ellipse detection algorithms due to the small scale of some of the ellipses and the difficulty of edge detection. The best result we obtained after adjusting the threshold for this sample is shown in the following figure.
![Figure.A](https://user-images.githubusercontent.com/40165523/184046892-76767816-c4cc-4dcf-a4f3-11590a5879d8.jpg)
Usually, for this case, we recommend to upsample the image twice or more and then detect it, there will be a slight improvement in such situation.

The detection of small ellipses has been a problem in this field, and we have done experiments for this. As can be seen from the figure, such methods will not be able to detect when the length of the long axis of a certain ellipse is less than 25 pixels.
![Figure.B](https://user-images.githubusercontent.com/40165523/184048282-991aafcb-577d-41c5-8d3e-3b4544183ad6.jpg)

In addition, we have also attached one of the more used images in the field and its detection results to demonstrate that our method works for circles.
![Figure.C](https://user-images.githubusercontent.com/40165523/184048543-778cec9c-204c-440e-baa1-a3d75ade69d4.png)

We would be honored if these descriptions could answer your questions.